File: code-coaches/medium/text-decompressor/text_decompressor.rb
def decompress_text(compressed_text)
  decompressed_text = ""
  (1...compressed_text.length).step(2) do |idx|
    decompressed_text += compressed_text[idx - 1] * compressed_text[idx].to_i
  end

  decompressed_text
end

def main
  compressed_text = gets.chomp
  puts decompress_text(compressed_text)
end

if __FILE__ == $0
  main
end

File: code-coaches/medium/text-decompressor/text_decompressor.cpp

#include <iostream>
#include <string>

std::string decompress_text(const std::string& compressed_text)
{
    std::string decompressed_text;
    for (int idx = 1; idx < compressed_text.length(); idx += 2)
    {
        char current_char = compressed_text[idx - 1];
        int repeat_count = compressed_text[idx] - '0';  // Convert character to integer
        decompressed_text.append(repeat_count, current_char);
    }

    return decompressed_text;
}

int main()
{
    std::string compressed_text;
    std::cin >> compressed_text;

    std::cout << decompress_text(compressed_text);

    return 0;
}
